subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -120,11 +122,31 @@
 import java.util.TreeSet;
 
 import static java.util.Collections.singletonMap;
+import static org.gradle.api.internal.artifacts.ArtifactAttributes.ARTIFACT_CLASSIFIER;
+import static org.gradle.api.internal.artifacts.ArtifactAttributes.ARTIFACT_EXTENSION;
+import static org.gradle.api.internal.artifacts.ArtifactAttributes.ARTIFACT_FORMAT;
 import static org.gradle.util.GUtil.addMaps;
 import static org.gradle.util.GUtil.isTrue;
 
 @NoConventionMapping
 public class DefaultProject extends AbstractPluginAware implements ProjectInternal, DynamicObjectAware {
+    private static final Action<AttributeMatchingStrategy<String>> ARTIFACT_ATTRIBUTE_CONFIG = new Action<AttributeMatchingStrategy<String>>() {
```

Comment:
Agreed, I didn't want to tackle this in this story (to be able to do gradle/performance#233 first). I'd like to have something merged before shuffling things around.